core/app/mailers/spree/base_mailer.rb

```diff
@@ -1,8 +1,8 @@
 module Spree
   class BaseMailer < ActionMailer::Base
 
-    def from_address
-      Spree::Config[:mails_from]
+    def from_address(store = nil)
+      store ? store.mail_from_address : Spree::Config[:mails_from]
```

Comment:
Should we consider putting a deprecation warning that store you should be passing store in to this method?